From f9288e2a62f7437a49045eea27a17a78352dfa43 Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Thu, 21 Aug 2014 04:13:36 -0400
Subject: [acpi] apei/cper: Fix status check during error printing

Message-id: <1408594432-28006-2-git-send-email-ruwang@redhat.com>
Patchwork-id: 88483
O-Subject: [PATCH RHEL7.1 BZ1076655 01/17] ACPI, APEI, CPER: Fix status check during error printing
Bugzilla: 1076655
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Chen, Gong" <gong.chen@linux.intel.com>

Backport of:
Original commit: 833ba4b1ba653f279c78e8e3352527acce8946e6
Original Author: "Chen, Gong" <gong.chen@linux.intel.com>
Original Date: Fri, 18 Oct 2013 14:27:51 -0700

 Commit aaf9d93be71c:
  ACPI / APEI: fix error status check condition for CPER
 only catches condition check before print, but a similar check is
 needed during printing CPER error sections.

Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/apei/cper.c b/drivers/acpi/apei/cper.c
index ef32f2d..572a28f 100644
--- a/drivers/acpi/apei/cper.c
+++ b/drivers/acpi/apei/cper.c
@@ -354,7 +354,7 @@ void apei_estatus_print(const char *pfx,
         cper_severity_str(severity));
  data_len = estatus->data_length;
  gdata = (struct acpi_hest_generic_data *)(estatus + 1);
- while (data_len > sizeof(*gdata)) {
+ while (data_len >= sizeof(*gdata)) {
   gedata_len = gdata->error_data_length;
   apei_estatus_print_section(pfx, gdata, sec_no);
   data_len -= gedata_len + sizeof(*gdata);
-- 
1.7.1